src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -177,13 +177,11 @@ public final void close() throws SQLException
    {
       if (delegate != ClosedConnection.CLOSED_CONNECTION) {
          leakTask.cancel();
+         closeOpenStatements();
```

Comment:
closeOpenStatements() accidentally moved out of try block ... can you move it back?